File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.cpp
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int ans = 0;
        bool dup = false;

        for (int i = 0; i < nums.size() - 1; ++i) {
            if (nums[i] == nums[i + 1]) {
                ans ^= nums[i];
                dup = true;
                ++i;
            }
        }

        return dup ? ans : 0;
    }
};

File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var duplicateNumbersXOR = function(nums) {
    let ans = 0

    for (let i = 0; i < nums.length - 1; i++) {
        if (nums.indexOf(nums[i], i + 1) != -1) {
            ans = ans ^ nums[i]
        }
    }

    return ans
};